tools/check-nuttx-release.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+set -e
+
+URL=$1
+TEMPDIR="dist.apache.org"
+ORIGINAL_DIR="$(pwd)"
+trap "rm -rf $TEMPDIR" EXIT
+
+function download_release() {
+    rm -rf "$TEMPDIR"
+    wget -r -np -R "index.html*" -P . --cut-dirs 7 "$URL"
+    cd "$TEMPDIR"
+}
+
+function check_sha512() {
+    # check release sha512
+    RELEASE_FILE=$1
+    echo "Checking $RELEASE_FILE sha512..."
+    sha512sum -c "$RELEASE_FILE.sha512"
+}
+
+function check_gpg() {
+    # check nuttx sha512 and gpg
+    RELEASE_FILE=$1
+    echo "Checking $RELEASE_FILE GPG signature:"
+    gpg --verify "$RELEASE_FILE.asc" "$RELEASE_FILE"
+    echo
+}
+
+function check_required_files() {
+    # check nuttx for required files
+    RELEASE_FILE=$1
+    RELEASE_DIR=$2 
+    rm -rf "$RELEASE_DIR"
+    tar xf "$RELEASE_FILE"
+    ERROR=0
+    if [ ! -f "$RELEASE_DIR/LICENSE" ]; then
+        echo "LICENSE file not present."
+        ERROR=1
+    fi
+    if [ ! -f "$RELEASE_DIR/NOTICE" ]; then
+        echo "NOTICE file not present."
+        ERROR=1
+    fi
+    if [ ! -f "$RELEASE_DIR/README.txt" ]; then
+        echo "README.txt file not present."
+        ERROR=1
+    fi
+    if [ ! -f "$RELEASE_DIR/DISCLAIMER-WIP" ]; then
+        echo "DISCLAIMER-WIP file not present."
+        ERROR=1
+    fi
+    if [ 0 -eq $ERROR ]; then
+        echo "OK: All required files exist."
+    fi
+}
+
+function check_nuttx() {
+    # check nuttx sha512 and gpg
+    RELEASE_FILE="$(ls *.tar.gz|head -1)"
+    check_sha512 "$RELEASE_FILE" check_gpg "$RELEASE_FILE" 
+    check_gpg "$RELEASE_FILE"
```

Comment:
```suggestion
    check_sha512 "$RELEASE_FILE"
    check_gpg "$RELEASE_FILE"
```
Was that a copy/paste typo?

Reply:
@Ouss4 yes, fixed.